examples/typescript/src/App.tsx

```diff
@@ -0,0 +1,35 @@
+import React, {useMemo} from 'react';
+import {locations} from "@contentful/app-sdk";
+import ConfigScreen from "./locations/ConfigScreen";
+import Field from "./locations/Field";
+import EntryEditor from "./locations/EntryEditor";
+import Dialog from "./locations/Dialog";
+import Sidebar from "./locations/Sidebar";
+import Page from "./locations/Page";
+import {useSDK} from "@contentful/react-apps-toolkit";
+
+const ComponentLocationSettings = {
+  [locations.LOCATION_APP_CONFIG]: <ConfigScreen />,
+  [locations.LOCATION_ENTRY_FIELD]: <Field />,
+  [locations.LOCATION_ENTRY_EDITOR]: <EntryEditor />,
+  [locations.LOCATION_DIALOG]: <Dialog />,
+  [locations.LOCATION_ENTRY_SIDEBAR]: <Sidebar />,
+  [locations.LOCATION_PAGE]: <Page />,
```

Comment:
Did you think about mapping to the components instead of already initializing the elements?

```typescript
const ComponentLocationSettings = {
  [locations.LOCATION_APP_CONFIG]: ConfigScreen,
  [locations.LOCATION_ENTRY_FIELD]: Field,
  [locations.LOCATION_ENTRY_EDITOR]: EntryEditor,
  [locations.LOCATION_DIALOG]: Dialog,
  [locations.LOCATION_ENTRY_SIDEBAR]: Sidebar,
  [locations.LOCATION_PAGE]: Page,
}
```

The behavior will be mostly the same, but the difference is that `React.createElement` is only called when the component is actually used. I'd find it the cleaner approach.

This needs a minor change below as well.